File: protobuf/protobuf3/Antlr4ng/Symbol.ts
import { TypeClassification } from './TypeClassification.js';

export class Symbol {
    name: string;
    classification: TypeClassification;
    members: Map<string, Symbol> = new Map();
    parent?: Symbol;

    constructor(name: string, classification: TypeClassification) {
        this.name = name;
        this.classification = classification;
    }

    toString(): string {
        let result = this.name;
        const classificationStr = TypeClassification[this.classification];
        result += " (with classification " + classificationStr;
        result += ")";
        if (this.parent) result += " of " + this.parent.toString();
        return result;
    }
}

File: c/Dart/Symbol.dart
import 'TypeClassification.dart';

class Symbol {
  String name = "";
  Set<TypeClassification> classification = {};
  Map<String, Symbol> members = {};
  Symbol? parent;

  @override
  String toString() {
    var result = name;
    var classificationStr = classification.map((c) => c.name).join(", ");
    result += " (with classification $classificationStr)";
    return result;
  }
}
